crypt4gh_middleware/middleware.py

```diff
@@ -33,15 +21,57 @@ def _add_decryption_executor(self):
                 "decrypt.py"
             ] + self.original_input_paths + [
                 "--output-dir",
-                self.output_dir
+                self.output_dir.as_posix()
             ]
         }
         self.request.json["executors"].insert(0, executor)
 
+    def _change_executor_paths(self):
+        """Change original input file paths in executors to the output directory."""
+        for executor_body in self.request.json["executors"]:
+            for i, path in enumerate(executor_body["command"]):
+                if path in self.original_input_paths:
+                    executor_body["command"][i] = (self.output_dir/path.split("/")[-1]).as_posix()
+
+    def _change_output_paths(self):
+        """Change original output file paths to the output directory if the output path is
+        the same as an input path.
+
+        Accounts for case where input file is modified in place in a TES request.
+        """
+        for output_body in self.request.json["outputs"]:
+            path = output_body["path"]
+            if path in self.original_input_paths:
+                output_body["path"] = (self.output_dir/path.split("/")[-1]).as_posix()
```

Comment:
Are we overriding the user's intent here or is this changed back again in some other place?

If the former, I tend to thing that it is better to handle things explicitly. In this case: If we don't allow modifying input files in place (which, IMO, is reasonable), then let's raise an error and tell the client about it (which is what we do for the volumes). The client can then change the output.

That way, we don't do what the user asked for. But at least we don't do what the user _did not_ ask for.